File: src/main/java/com/robohorse/robopojogenerator/generator/utils/InnerObjectResolver.java
package com.robohorse.robopojogenerator.generator.utils;

import com.robohorse.robopojogenerator.generator.consts.ClassType;
import org.json.JSONArray;
import org.json.JSONObject;

/**
 * Created by vadim on 23.09.16.
 */
public abstract class InnerObjectResolver {

    private ClassGenerateHelper classGenerateHelper = new ClassGenerateHelper();

    public void resolveClassType(Object object, String key) {
        if (object instanceof JSONObject) {
            onJsonObjectIdentified(classGenerateHelper.getClassName(key));

        } else if (object instanceof JSONArray) {
            onJsonArrayIdentified(key);

        } else if (object instanceof String) {
            onInnerObjectIdentified(ClassType.STRING);

        } else if (object instanceof Integer) {
            onInnerObjectIdentified(ClassType.INTEGER);

        } else if (object instanceof Double) {
            onInnerObjectIdentified(ClassType.DOUBLE);

        } else if (object instanceof Float) {
            onInnerObjectIdentified(ClassType.FLOAT);

        } else if (object instanceof Long) {
            onInnerObjectIdentified(ClassType.LONG);

        } else if (object instanceof Boolean) {
            onInnerObjectIdentified(ClassType.BOOLEAN);

        } else {
            onInnerObjectIdentified(ClassType.OBJECT);
        }
    }

    public abstract void onInnerObjectIdentified(ClassType classType);

    public abstract void onJsonObjectIdentified(String classType);

    public abstract void onJsonArrayIdentified(String classType);
}

File: generator/src/main/kotlin/com/robohorse/robopojogenerator/processing/InnerObjectResolver.kt
package com.robohorse.robopojogenerator.processing

import com.robohorse.robopojogenerator.properties.ClassEnum
import com.robohorse.robopojogenerator.properties.ClassEnum.BOOLEAN
import com.robohorse.robopojogenerator.properties.ClassEnum.DOUBLE
import com.robohorse.robopojogenerator.properties.ClassEnum.FLOAT
import com.robohorse.robopojogenerator.properties.ClassEnum.INTEGER
import com.robohorse.robopojogenerator.properties.ClassEnum.LONG
import com.robohorse.robopojogenerator.properties.ClassEnum.OBJECT
import com.robohorse.robopojogenerator.properties.ClassEnum.STRING
import org.json.JSONArray
import org.json.JSONObject

internal abstract class InnerObjectResolver {

    fun resolveClassType(targetItem: Any) {
        when (targetItem) {
            is JSONObject -> onJsonObjectIdentified()
            is JSONArray -> onJsonArrayIdentified()
            is String -> onInnerObjectIdentified(STRING)
            is Int -> onInnerObjectIdentified(INTEGER)
            is Double -> onInnerObjectIdentified(DOUBLE)
            is Float -> onInnerObjectIdentified(FLOAT)
            is Long -> onInnerObjectIdentified(LONG)
            is Boolean -> onInnerObjectIdentified(BOOLEAN)
            else -> onInnerObjectIdentified(OBJECT)
        }
    }

    abstract fun onInnerObjectIdentified(classEnum: ClassEnum?)

    abstract fun onJsonObjectIdentified()

    abstract fun onJsonArrayIdentified()
}
